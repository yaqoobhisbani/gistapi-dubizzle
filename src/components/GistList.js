import styled from "styled-components";
import React from "react";
import Gist from "./Gist";

const GistList = ({ gists }) => {
  return (
    <Wrapper>
      {Array.isArray(gists) &&
        gists.length > 0 &&
        gists.map((gist) => <Gist gist={gist} key={gist.id} />)}
    </Wrapper>
  );
};

const Wrapper = styled.div`
  width: 50%;
  display: flex;
  flex-direction: column;
  align-self: center;

  @media (max-width: 960px) {
    width: 80%;
  }

  @media (max-width: 768px) {
    width: 90%;
  }
`;

export default React.memo(GistList);
