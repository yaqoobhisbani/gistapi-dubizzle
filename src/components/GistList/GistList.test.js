import React from "react";
import { render } from "@testing-library/react";
import GistList from "./GistList";

describe("GistList", () => {
  const gists = [
    {
      id: "1",
      description: "Gist 1",
      files: {},
      owner: {
        login: "user1",
        avatar_url: "https://example.com/avatar1.jpg",
      },
    },
    {
      id: "2",
      description: "Gist 2",
      files: {},
      owner: {
        login: "user2",
        avatar_url: "https://example.com/avatar2.jpg",
      },
    },
  ];

  test("matches snapshot", () => {
    const { container } = render(<GistList gists={gists} />);
    expect(container).toMatchSnapshot();
  });
});
