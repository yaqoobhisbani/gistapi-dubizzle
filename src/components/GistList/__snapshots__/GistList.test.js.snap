// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GistList matches snapshot 1`] = `
<div>
  <div
    class="sc-irTswW eiEcqt"
  >
    <div
      class="sc-fsQiph hGyyWv"
    >
      <div
        class="sc-hLseeU BbBzW"
      >
        <div
          class="sc-eDDNvR idlsVM"
        >
          <img
            class="sc-jTrPJq dYZuXy"
            src="https://example.com/avatar1.jpg"
          />
          <a
            class="sc-gLDzan dAgnKx"
          >
            user1
          </a>
        </div>
        <div
          class="sc-iAEyYk fIvPjx"
        >
          <div
            class="sc-gueYoa cHbybZ"
            style="margin-right: 16px;"
          >
            <span
              class="octicon octicon-code"
              data-testid="icon"
              style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
            />
            <a
              class="sc-dmqHEX lmtyIK"
            >
              0 Files
            </a>
          </div>
          <div
            class="sc-gueYoa cHbybZ"
            style="margin-right: 16px;"
          >
            <span
              class="octicon octicon-repo-forked"
              data-testid="icon"
              style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
            />
            <a
              class="sc-dmqHEX lmtyIK"
            >
              Forks
            </a>
          </div>
          <div
            class="sc-gueYoa cHbybZ"
            style="margin-right: 16px;"
          >
            <span
              class="octicon octicon-comment"
              data-testid="icon"
              style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
            />
            <a
              class="sc-dmqHEX lmtyIK"
            >
              Comments
            </a>
          </div>
          <div
            class="sc-gueYoa cHbybZ"
          >
            <span
              class="octicon octicon-star"
              data-testid="icon"
              style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
            />
            <a
              class="sc-dmqHEX lmtyIK"
            >
              Stars
            </a>
          </div>
        </div>
      </div>
      <div
        class="sc-qRumB ftafra"
      >
        <p
          class="sc-beqWaB yDHQF"
        >
          <span>
            Created at: 
          </span>
          <span>
            NaN/NaN/NaN
          </span>
        </p>
        <p
          class="sc-beqWaB yDHQF"
          style="margin-left: 16px;"
        >
          <span>
            Last updated: 
          </span>
          <span>
            NaN/NaN/NaN
          </span>
        </p>
      </div>
      <p
        class="sc-kFuwaP hgxAcK"
      >
        Gist 1
      </p>
      <div
        class="sc-qRumB sc-jsMahE ftafra dTKjie"
      />
    </div>
    <div
      class="sc-fsQiph hGyyWv"
    >
      <div
        class="sc-hLseeU BbBzW"
      >
        <div
          class="sc-eDDNvR idlsVM"
        >
          <img
            class="sc-jTrPJq dYZuXy"
            src="https://example.com/avatar2.jpg"
          />
          <a
            class="sc-gLDzan dAgnKx"
          >
            user2
          </a>
        </div>
        <div
          class="sc-iAEyYk fIvPjx"
        >
          <div
            class="sc-gueYoa cHbybZ"
            style="margin-right: 16px;"
          >
            <span
              class="octicon octicon-code"
              data-testid="icon"
              style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
            />
            <a
              class="sc-dmqHEX lmtyIK"
            >
              0 Files
            </a>
          </div>
          <div
            class="sc-gueYoa cHbybZ"
            style="margin-right: 16px;"
          >
            <span
              class="octicon octicon-repo-forked"
              data-testid="icon"
              style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
            />
            <a
              class="sc-dmqHEX lmtyIK"
            >
              Forks
            </a>
          </div>
          <div
            class="sc-gueYoa cHbybZ"
            style="margin-right: 16px;"
          >
            <span
              class="octicon octicon-comment"
              data-testid="icon"
              style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
            />
            <a
              class="sc-dmqHEX lmtyIK"
            >
              Comments
            </a>
          </div>
          <div
            class="sc-gueYoa cHbybZ"
          >
            <span
              class="octicon octicon-star"
              data-testid="icon"
              style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
            />
            <a
              class="sc-dmqHEX lmtyIK"
            >
              Stars
            </a>
          </div>
        </div>
      </div>
      <div
        class="sc-qRumB ftafra"
      >
        <p
          class="sc-beqWaB yDHQF"
        >
          <span>
            Created at: 
          </span>
          <span>
            NaN/NaN/NaN
          </span>
        </p>
        <p
          class="sc-beqWaB yDHQF"
          style="margin-left: 16px;"
        >
          <span>
            Last updated: 
          </span>
          <span>
            NaN/NaN/NaN
          </span>
        </p>
      </div>
      <p
        class="sc-kFuwaP hgxAcK"
      >
        Gist 2
      </p>
      <div
        class="sc-qRumB sc-jsMahE ftafra dTKjie"
      />
    </div>
  </div>
</div>
`;
